edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/storage/MessageStore.cs

```diff
@@ -232,96 +234,124 @@ void EnsureCleanupTask(object state)
             /// // TODO - Update cleanup logic to cleanup expired messages from entity store as well.
             /// </summary>
             async Task CleanupMessages()
+            {
+                try
+                {
+                    await this.CleanQueue(this.checkEntireQueueOnCleanup);
+                }
+                catch (Exception ex)
+                {
+                    Events.ErrorCleaningMessages(ex);
+                    throw;
+                }
+            }
+
+            private async Task CleanQueue(bool checkEntireQueueOnCleanup)
             {
                 long totalCleanupCount = 0;
                 long totalCleanupStoreCount = 0;
-
-                try
```

Comment:
The diff looks a bit funky here because I moved the queue cleaning logic into a method